keyboards/4pplet/yakiimo/proto/matrix.c

```diff
@@ -0,0 +1,178 @@
+    palSetPadMode(GPIOB, 14, PAL_MODE_INPUT_PULLDOWN);
+    palSetPadMode(GPIOB, 12, PAL_MODE_INPUT_PULLDOWN);
+    palSetPadMode(GPIOB, 9, PAL_MODE_INPUT_PULLDOWN);
+    palSetPadMode(GPIOB, 8, PAL_MODE_INPUT_PULLDOWN);
+    palSetPadMode(GPIOB, 5, PAL_MODE_INPUT_PULLDOWN);
+    palSetPadMode(GPIOB, 4, PAL_MODE_INPUT_PULLDOWN);
+    palSetPadMode(GPIOA, 15, PAL_MODE_INPUT_PULLDOWN);
+    palSetPadMode(GPIOB, 3, PAL_MODE_INPUT_PULLDOWN);
+
+    memset(matrix, 0, MATRIX_ROWS * sizeof(matrix_row_t));
+    memset(matrix_debouncing, 0, MATRIX_COLS * sizeof(matrix_row_t));
+
+    matrix_init_quantum();
+}
+
+uint8_t matrix_scan(void) {
+    // actual matrix
+    for (int col = 0; col < MATRIX_COLS; col++) {
+        matrix_row_t data = 0;
+        switch (col) {
+            case 0:
+                palSetPad(GPIOB, 0);
+                break;
+            case 1:
+                palSetPad(GPIOA, 5);
+                break;
+            case 2:
+                palSetPad(GPIOA, 4);
+                break;
+            case 3:
+                palSetPad(GPIOA, 3);
+                break;
+            case 4:
+                palSetPad(GPIOA, 2);
+                break;
+            case 5:
+                palSetPad(GPIOA, 1);
+                break;
+            case 6:
+                palSetPad(GPIOA, 0);
+                break;
+            case 7:
+                palSetPad(GPIOC, 15);
+                break;
+            case 8:
+                palSetPad(GPIOA, 9);
+                break;
+        }
+
+        // need wait to settle pin state
+        wait_us(20);
+        // read row data { B10, B1, C13, C14, B14, B12, B9, B8, B5, B4, A15, B3  }
+        data = ((palReadPad(GPIOB, 10) << 0) | (palReadPad(GPIOB, 1) << 1) | (palReadPad(GPIOC, 13) << 2) | (palReadPad(GPIOC, 14) << 3) | (palReadPad(GPIOB, 14) << 4) | (palReadPad(GPIOB, 12) << 5) | (palReadPad(GPIOB, 9) << 6) | (palReadPad(GPIOB, 8) << 7) | (palReadPad(GPIOB, 5) << 8) | (palReadPad(GPIOB, 4) << 9) | (palReadPad(GPIOA, 15) << 10) | (palReadPad(GPIOB, 3) << 11));
+        // unstrobe  col { B0, A5, A4, A3, A2, A1, A0, C15, A9 }
+        switch (col) {
+            case 0:
+                palClearPad(GPIOB, 0);
+                break;
+            case 1:
+                palClearPad(GPIOA, 5);
+                break;
+            case 2:
+                palClearPad(GPIOA, 4);
+                break;
+            case 3:
+                palClearPad(GPIOA, 3);
+                break;
+            case 4:
+                palClearPad(GPIOA, 2);
+                break;
+            case 5:
+                palClearPad(GPIOA, 1);
+                break;
+            case 6:
+                palClearPad(GPIOA, 0);
+                break;
+            case 7:
+                palClearPad(GPIOC, 15);
+                break;
+            case 8:
+                palClearPad(GPIOA, 9);
+                break;
+        }
+
+        if (matrix_debouncing[col] != data) {
```

Comment:
This should use `CUSTOM_MATRIX = lite` to avoid reimplementing debounce/print/etc/

Reply:
I updated the code to use CUSTOM_MATRIX = lite without print. I'm not sure how I could do debounce differently. This is mainly just me following Jack Humberts implementation, so I'd be super happy for tips/suggestions :)

Reply:
custom matrix lite handles debounce through an API.  It's completely abstracted from the lite code, so there is no need to implement any sort of debouncing. 

However, if you need custom debouncing for the boards matrix, that can/should be implemented separately, through the debounce API:

https://docs.qmk.fm/#/feature_debounce_type

Reply:
Thanks for the input! I removed the proto code for now, I'll check this out and try to implement it differently :)